### DISCUSSION TITLE
VM resources for Leaf installation

### DISCUSSION THREAD

**Discussion by artgoldberg** [uwrit/leaf#401] [Q&A] [answered]
Hi Nic & Team

We plan to deploy Leaf on a Centos 7.6 VM. We may deploy it directly on th OS, or we may run the [Leaf Docker deployment](https://github.com/uwrit/leaf/discussions/388) on it.

The CDM being used has about 8 million patients, 98 million visits, 115 million diagnoses, and 355 million test results. I expect at most 20 or 30 concurrent Leaf users.

How should we configure the VM resources, RAM and cores for the installation?

Thanks
Arthur

**Comment by ndobb** [comment]
@jprosser or @mh2727, can you please advise based on our experience with Leaf deployments?

Also for clarification, @artgoldberg, will this VM be hosting _both_ the Leaf API and CDM database, or only the API? (I'm guessing both based on description but just to be sure)

Best,
-nic

**Reply by artgoldberg** [reply]
Thanks @ndobb 

This VM will host just the Leaf API server, running in a Docker container. The CDM and Leaf databases will be in another VM (on the same host) running SQL Server. I plan to initialize the Leaf DB by running the scripts in db/build and the OMOP leaf-scripts, also in a Docker container.

Regards, Arthur

**Comment by mh2727** [comment, marked as answer]
@artgoldberg 
We generally do the following for our Leaf deployments.

- App vm
    - RAM: 2GB
    - CPU: 2
    - Disk: 10GB

- DB vm
    - RAM: 4GB
    - CPU: 2
    - Disk: depends on the size of your database

- Web vm
    - RAM: 2GB
    - CPU: 2
    - Disk: 10GB

We can also share our Dockerfile we use to deploy the Leaf app if that would be helpful.

--
Mehadi Hassan

**Reply by artgoldberg** [reply]
Thanks @mh2727 , that's helpful. And I would be interested in seeing your Dockerfile.

Arthur

**Reply by mh2727** [reply]
@artgoldberg
Retracting email pasted here. Apologies.

--
Mehadi Hassan

**Reply by artgoldberg** [reply]
Yes. Thanks @mh2727 .
